### PR TITLE
PSU Units

### LINKED ISSUE #40: PSU Conversion
We should add a general purpose PSU unit converter to avoid salinity not being a real "unit"

Currently, converting `psu -> 0.001`:
```log
pymorize.units:handle_unit_conversion:120 - Converting units: psu -> 0.001
```
Gives:
```log
ValueError: Cannot convert variables:
    incompatible units for variable 'so': Unit expression cannot have a scaling factor.
```

See more ideas [here](https://stackoverflow.com/questions/39153885/how-to-define-and-use-percentage-in-pint).

This needs to go in `src/pymorize/units.py` as a general case added to the unit registry.

### PR DESCRIPTION
**Pull request by pgierz**
Adds the ability to convert PSU units in Pint.

Will close #40

### CODE REVIEW

File: tests/unit/test_units.py [hunk 1]
```diff
@@ -156,3 +156,11 @@ def test_when_tartget_unit_is_empty_string_raises_error(rule_with_units, from_un
     da = xr.DataArray(10, attrs={"units": from_unit})
     with pytest.raises(ValueError):
         handle_unit_conversion(da, rule_spec)
+
+
+def test_units_with_psu():
+    da = xr.DataArray(10, attrs={"units": "psu"})
+    new_da = da.pint.quantify("psu")
+    new_da = new_da.pint.to("0.001").pint.dequantify()
```
Comment (siligam) [inline tests/unit/test_units.py:164]: `pint.to` does not take any scalar values, needs to be a valid unit name

File: tests/unit/test_units.py [hunk 2]
```diff
@@ -156,3 +156,11 @@ def test_when_tartget_unit_is_empty_string_raises_error(rule_with_units, from_un
...
+    assert new_da.data == np.array(10)
+    assert new_da.attrs.get("units") == "0.001"
```
Comment (siligam) [inline tests/unit/test_units.py:166]: if the intension is to rename units without changing the magnitude, then da.attrs['units'] = "0.001" is the way to go.

Comment (pgierz) [reply tests/unit/test_units.py:166]: No, the point is to demonstrate that this is what your code automatically tries to do in `units.py`, and leads to a crash. So, we need to have a way to handle this case. You could abstract this test into one that uses the actual unit conversion function, and then handles the conversion correctly.

File: src/pymorize/units.py [hunk 3]
```diff
@@ -34,6 +38,23 @@
...
+# ureg.define("0.001 = ppt")
+
+
+@xr.register_dataarray_accessor("pint")
+class PSUAwarePintDataArrayAccessor(pint_xarray.accessors.PintDataArrayAccessor):
+    def to(self, unit):
+        if unit == "0.001":
+            logger.critical(">>>> PG PART!!!!")
+            return super().to("ppt")
+        return super().to(unit)
```
Comment (pgierz) [inline src/pymorize/units.py:52]: @siligam: this is what I had in mind, but you would still need to fix it. Unfortunately, `ppt` (should be parts per thousand) is already defined as `picopint`.

I don't think we should override a built-in `pint` unit at this point; rather just handle the conversion logic manually. This conversion override is where you should probably include the corrected code.

Comment (pgierz) [reply src/pymorize/units.py]: This is what this looks like in practice:

```python
>>> import xarray as xr
>>> da = xr.DataArray([10.0], attrs={"units": "psu"})  # Comes out of FESOM
>>> # The above is equivalent to:
>>> ds = xr.open_dataset("/my/fesom/file.nc")  # Attribute sst is the variable name
>>> da = ds.sst
>>> da.pint.quantify() # Turns it into a "unit-aware" array
>>> target_unit = "0.001"
>>> new_da = da.pint.to(target_unit)
```

**Comment by pgierz** [comment]
Superseded by #55.

**Pull request closed without merging**